test/StdChains.t.sol

```diff
@@ -25,6 +24,16 @@ contract StdChainsTest is Test {
         }
     }
 
+    function testCannotSetChain_ChainIdExists() public {
+        setChain({key: "custom_chain", name: "Custom Chain", chainId: 123456789, rpcUrl: "https://custom.chain/"});
+
+        vm.expectRevert(
+            'StdChains setChain(string,string,uint256,string): Chain ID 123456789 already used by "custom_chain"'
+        );
```

Comment:
Does this mean users can't change the name of the default `Anvil` chain to `Hardhat`?

Reply:
Name, as in `Chain.name`? Yes.

Reply:
What they cannot do is add `"hardhat"` Chain that has the same chain ID as `"anvil"`.

Reply:
This lets you:

- add a new Chain (must have unique chain ID)
- modify an existing chain

Reply:
If good merge.

Reply:
Great, thanks!